ICSharpCode.Decompiler/Disassembler/DisassemblerHelpers.cs

```diff
@@ -317,7 +319,7 @@ public static string EscapeString(string str)
 						// print control characters and uncommon white spaces as numbers
 						if (char.IsControl(ch) || char.IsSurrogate(ch) || (char.IsWhiteSpace(ch) && ch != ' '))
 						{
-							sb.Append("\\u" + ((int)ch).ToString("x4"));
+							sb.AppendFormat("\\u{0:x4}", ch);
```

Comment:
This change introduces a bug. There is a good reason for the cast to `int` here: `char.ToString()` is not the same as `int.ToString()`.

See this for example:

```
> string.Format("\\u{0:x4}", '\u0041')
"\\uA"
> string.Format("\\u{0:x4}", (int)'\u0041')
"\\u0041"
```

![image](https://user-images.githubusercontent.com/344204/119791747-64008800-bed5-11eb-87c2-9127aff30628.png)


```
System.Text.EncoderFallbackException: 'Unable to translate Unicode character \uDB80 at index 309 to specified code page.'
```

Other than that your changes seem fine...

Reply:
Thanks, will fix that and rebase to master.